kedro/framework/cli/micropkg.py

```diff
@@ -47,6 +50,50 @@
 """
 
 
+class _EquivalentRequirement(Requirement):
```

Comment:
Maybe we should be adding some docstrings here to describe the purpose of the class and its methods.

Reply:
I'll add a docstring to make it more clear that this essentially overrides `packaging.requirements.Requirement.__eq__` to be backwards compatible with `pkg_resources`